### PR TITLE
Improve grpc-xds javadoc and make it publishable

### PR DESCRIPTION
**Pull request by dapengzhang0**
- Improve package-info.java and make minor changes to other javadoc.
- Make Orca API non-public for the moment.
- Make grpc-xds publishable.

### CODE REVIEW

**Comment by voidzcy** [comment]
Does this matter? We don't have any public API in xds package though.

**Comment by dapengzhang0** [comment]
If the providers are moved to internal package, then everything has to be moved to internal package, which seems fine only that every class has a long package name. Btw why all Orca* classes are public `@Experimental` not internal?

**Comment by voidzcy** [comment]
> If the providers are moved to internal package, then everything has to be moved to internal package, which seems fine only that every class has a long package name. Btw why all Orca* classes are public `@Experimental` not internal?

They should be internal as well. We should remove the `@Experimental`. Everything should be internal. No Javadoc should be shown for xds.

**Comment by dapengzhang0** [comment]
PTAL again

File: xds/src/main/java/io/grpc/xds/XdsNameResolverProvider.java [hunk 1]
```diff
@@ -35,6 +36,7 @@
  * through {@link io.grpc.NameResolverRegistry#asFactory#newNameResolver(URI, Args)} with the URI
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/XdsNameResolverProvider.java]: Can you remove this paragraph? This tag is causing javadoc warning.

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/XdsNameResolverProvider.java]: Fixed.

File: xds/src/main/java/io/grpc/xds/package-info.java [hunk 2]
```diff
@@ -15,7 +15,16 @@
  */
 
 /**
- * The XDS loadbalancer plugin implementation.
+ * Library for gPRC proxyless service mesh using Envoy xDS protocol.
+ *
+ * <p>The package currently includes a name resolver plugin and a family of load balancer plugins.
+ * A gRPC channel for a target with {@code "xds-experimental"} scheme will load the plugins and a
+ * bootstrap file and communicate with an external control plane management server (e.g. Traffic
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/package-info.java]: This sentence may need to be rephrased. It's hard to read.

File: xds/src/main/java/io/grpc/xds/package-info.java [hunk 3]
```diff
@@ -19,9 +19,9 @@
...
  * <p>The package currently includes a name resolver plugin and a family of load balancer plugins.
  * A gRPC channel for a target with {@code "xds-experimental"} scheme will load the plugins and a
- * bootstrap file and communicate with an external control plane management server (e.g. Traffic
- * Director) that speaks Envoy xDS protocol to retrieve routing, load balancing, load reporting
- * configurations etc. for the channel. More features will be added.
+ * bootstrap file, and will communicate with an external control plane management server (e.g.
+ * Traffic Director) that speaks Envoy xDS protocol to retrieve routing, load balancing, load
+ * reporting configurations etc. for the channel. More features will be added.
  *
  * <p>The library is currently in an agile development phase, so API and design are subject to
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/package-info.java]: This sentence doesn't look quite grammatically correct.

**Comment by voidzcy** [comment]
We can merge this first so that the artifact is publishable. I can create another PR to polish/improve the documentation after this.

**Pull request merged**